coderd/database/dbauthz/dbauthz.go

```diff
@@ -1173,6 +1173,20 @@ func (q *querier) GetTemplateDAUs(ctx context.Context, arg database.GetTemplateD
 	return q.db.GetTemplateDAUs(ctx, arg)
 }
 
+func (q *querier) GetTemplateDailyInsights(ctx context.Context, arg database.GetTemplateDailyInsightsParams) ([]database.GetTemplateDailyInsightsRow, error) {
+	if err := q.authorizeContext(ctx, rbac.ActionRead, rbac.ResourceSystem); err != nil {
```

Comment:
You could argue this should be READ rbac.ResourceTemplate

Reply:
I will admit I don't 100% understand this layer or what should be done when and why. I was following previous precedent of `GetTemplateDAUs` (which is also for admins only). Happy to make any changes you deem necessary.

Reply:
Folks can argue about this in a follow-up, we can leave it as ResourceSystem for now.